libs/dotnet-sdk/Registration/RegisterFeatureBoard.cs

```diff
@@ -0,0 +1,76 @@
+using Featureboard.DotnetSdk.Models;
+using Featureboard.DotnetSdk.State;
+using Featureboard.DotnetSdk.UpdateStrategies;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Featureboard.DotnetSdk.Registration;
+
+public static class RegisterFeatureBoard
+{
+  private static UpdateStrategy updateStrategy = UpdateStrategy.None;
+  public static FeatureBoardBuilder AddFeatureBoard<TFeatures, TAudienceProvider>(this IServiceCollection services) where TFeatures : class, IFeatures where TAudienceProvider : class, IAudienceProvider
+  {
+    var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+    services.AddOptions<FeatureboardOptions>().Bind(configuration.GetSection(nameof(FeatureboardOptions))).ValidateDataAnnotations();
+
+    services.AddHttpClient(Constants.HttpClientName, (serviceProvider, client) =>
+    {
+      var options = serviceProvider.GetRequiredService<IOptions<FeatureboardOptions>>();
+      client.BaseAddress = options.Value.HttpEndpoint;
+      client.DefaultRequestHeaders.Add("x-environment-key", options.Value.EnvironmentApiKey);
+      client.Timeout = TimeSpan.FromMilliseconds(options.Value.MaxAgeMs - 1);
+    });
+
+    services.AddScoped<IFeatureBoardClient<TFeatures>, FeatureBoardClient<TFeatures>>();
+    services.AddScoped<IAudienceProvider, TAudienceProvider>();
+    services.AddSingleton<IFeatureboardState, FeatureboardState>();
+
+    return new FeatureBoardBuilder(services);
+  }
+
+  public static FeatureBoardBuilder WithOnRequestUpdateStrategy(this FeatureBoardBuilder builder)
+  {
+    builder.Services.AddSingleton<IFeatureboardHttpClient, FeatureboardHttpClient>();
+    updateStrategy = UpdateStrategy.OnRequest;
+
+    return builder;
+  }
+
+  public static FeatureBoardBuilder WithPollingUpdateStrategy(this FeatureBoardBuilder builder)
+  {
+    if (updateStrategy != UpdateStrategy.None)
+      throw new ApplicationException("You can only have one update strategy registered");
+    builder.Services.AddSingleton<IFeatureboardHttpClient, FeatureboardHttpClient>();
```

Comment:
I think the same `IFeatureboardHttpClient` registration is being added in both `WIthXXXStrategy()` methods, so move in to `AddFeaturesBoard()` instead?

Reply:
I've added it here as methods such as websockets won't make use of it.